plugins/javaembed/javaembed.cpp

```diff
@@ -1809,14 +1813,13 @@ class ECLDatasetIterator : public CInterfaceOf<IInterface>
       javaBuilder(JNIenv, &dummyField, className)
     {
         nextRead = false;
-        nextPending = NULL;
     }
 
     bool hasNext()
     {
         if (!nextRead)
         {
-            nextPending = (const byte *) val->ungroupedNextRow();
+            nextPending.setown(val->ungroupedNextRow());
```

Comment:
This was causing newly-added tests to fail, as they revealed a memory leak when passing datasets in via iterator.

Reply:
(not a new issue)